### DISCUSSION TITLE
Error “Predictor not found” when using AI Agent with LangChain Conversational Model

### DISCUSSION THREAD

**Discussion by SimonJove** [mindsdb/mindsdb#10229] [Get help]
I am encountering an issue while trying to create and use an AI Agent following the instructions from the documentation [Build AI Agents](https://docs.mindsdb.com/use-cases/ai_agents/build_ai_agents). Despite following the steps closely, I receive the error “Predictor not found: name=‘demo_model’ id=’-’” when querying the agent. Below are the detailed steps I performed:

Step 1: Create a Conversational Model

I created a conversational model using the following SQL:

CREATE MODEL demo_model
PREDICT answer
USING
    engine = 'langchain',
    openai_api_key = 'sk-xxxxxx',
    model_name = 'gpt-4',
    mode = 'conversational',
    user_column = 'question',
    assistant_column = 'answer',
    max_tokens = 100,
    temperature = 0,
    verbose = True,
    prompt_template = 'Answer the user input in helpful way: {{questions}}';

The model creation command executed successfully without any errors.

Step 2: Create a Skill

I created a skill for interacting with a MySQL database:

CREATE SKILL text_to_sql_skill_for_sys_user
USING
    type = 'text_to_sql',
    database = 'mysql_newyear',
    tables = ['sys_user'],
    description = "This is user infos data";

Step 3: Create an AI Agent

I created an agent combining the conversational model and the skill:

CREATE AGENT text_to_sql_for_sys_user_agent USING
    model = 'demo_model',
    skills = ['text_to_sql_skill_for_sys_user'];

Step 4: Query the Agent

When I tried querying the agent:

SELECT * FROM text_to_sql_for_sys_user_agent
WHERE question = "how many users in sys_user?";

I encountered the following error:

Predictor not found: name='demo_model' id='-'

Environment

	•	MindsDB Version: 24.11.4.1
	•	Gui_version: 4.42
	•	Host: Docker Environment

Questions

	1.	Why does the system say that the predictor “demo_model” is not found, even though the model creation command executed successfully?
	2.	Are there additional steps or configurations I might have missed for using the langchain engine in MindsDB?
	3.	Could this error be related to specific settings in the Docker environment or MindsDB Studio?

Thank you for your help! Let me know if you need further information or logs to debug this issue.

**Comment by jiashengguo** [comment]
Got the same issue
